### PR TITLE
[Ut] Add drop meta unit test and remove some redundant code in Alter

### LINKED ISSUE #5331: [Bug] when force drop db, table, partition, related metas such as tablets are not erased
Sometimes, we need force drop meta and not need recycle db, table, partition anymore, but now, when we drop meta forcely, the related meta such as tablet is not erased.

### PR DESCRIPTION
**Pull request by caiconghui**
## Proposed changes

Add some unit test for drop db, drop table and drop partition

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have created an issue on (Fix #5331) and described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If these changes need document changes, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java [hunk 1]
```diff
@@ -230,16 +230,6 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
         List<AlterClause> alterClauses = Lists.newArrayList();
         // some operations will take long time to process, need to be done outside the table lock
         boolean needProcessOutsideTableLock = false;
-
-        // check conflict alter ops first
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:234]: Why do we needn't check conflict alter ops first?

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:234]: > Why do we needn't check conflict alter ops first?

alterClauses is always empty here, so check conflict alter ops makes no sense here.  in processAlterOlapTable and processAlterExternalTable will check conflict alter ops

Comment (kangkaisen) [reply fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:234]: Ok, Thanks.

**Review by kangkaisen — APPROVED** [review]
+1

**Pull request merged**